### ISSUE TITLE
ipv6 getaddrinfo error

### ISSUE THREAD

**Issue by changdingfang** [warmcat/libwebsockets#3286]
error log: "lws_getaddrinfo46: getaddrinfo 'ipv6 address' says 7"
error log:  lws_client_connect_3_connect: dns lookup failed 7

In the lws_client_connect_2_dnsreq interface, an error is reported when the getaddrinfo command is executed for the IPv6 address in square brackets. As a result, the link cannot be set up.

Android machine

**Comment by lws-team** [comment]
So... this works OK without the square brackets?

**Comment by changdingfang** [comment]
yes

**Comment by changdingfang** [comment]
When the lws_client_connect_via_info interface is invoked, the addr field of lws_client_connect_info does not contain square brackets. As a result, the HTTP part fails.

**Comment by lws-team** [comment]
It sounds like some code is needed to strip the square brackets if present before using it with things that don't know about it.

**Comment by changdingfang** [comment]
Yes, the square brackets are used to isolate IPv6 addresses and ports, but some interfaces cannot recognize the square brackets and need to be removed separately.

**Comment by changdingfang** [comment]
Can this problem be handled in later versions?

**Comment by changdingfang** [comment]
What is the network address of the parameter transferred by the lws_client_connect_via_info interface?
If there are no square brackets, the libwebsocket library cannot identify the IPv6 address.
If the content contains square brackets, the libwebsocket library intercepts the content and identifies it as a URL.
I understand that this is a problem that the libwebsocket library is not properly adapted.

**Comment by lws-team** [comment]
> I understand that this is a problem that the libwebsocket library is not properly adapted.

Well, it's quite possible, since I mostly wrote the whole thing over 15 years alone, unpaid.  But in this case, I think you don't understand the problem yet.

Are you using lws_parse_url()?  As I wrote above your comment, [] is not an ipv6 thing, it is a URL thing.  The api for dealing with it in lws lives in the URL handling api.
